xm-commons-domainevent/src/main/java/com/icthh/xm/commons/domainevent/config/WebApiSource.java

```diff
@@ -0,0 +1,236 @@
+            .operation(getOperation(request.getMethod()))
+            .msName(appName)
+            .source(DefaultDomainEventSource.HTTP.name())
+            .userKey(userKey)
+            .clientId(clientId)
+            .tenant(tenant)
+            .payload(createPayload(request, response, requestBody, responseBody))
+            .build();
+    }
+
+    private HttpDomainEventPayload createPayload(HttpServletRequest request, HttpServletResponse response,
+                                                 String requestBody, String responseBody) {
+
+        HttpDomainEventPayload payload = new HttpDomainEventPayload();
+        payload.setMethod(request.getMethod());
+        payload.setUrl(request.getRequestURI());
+        payload.setQueryString(request.getQueryString());
+        payload.setRequestLength(request.getContentLengthLong());
+        payload.setRequestBody(maskContent(requestBody, request.getRequestURI(), true, request.getMethod()));
+        payload.setResponseBody(maskContent(responseBody, request.getRequestURI(), false, request.getMethod()));
+        payload.setResponseLength((long) responseBody.length());
+        payload.setRequestHeaders(getRequestHeaders(request));
+        payload.setResponseHeaders(getResponseHeaders(response));
+        payload.setResponseCode(response.getStatus());
+        payload.setExecTime(MdcUtils.getExecTimeMs());
+        return payload;
+    }
+
+    private HttpHeaders getRequestHeaders(HttpServletRequest request) {
+        Map<String, List<String>> headers = new LinkedHashMap<>();
+        Enumeration<String> names = request.getHeaderNames();
+        while (names.hasMoreElements()) {
+            String name = names.nextElement();
+            if (!EXCLUDE_HEADERS.contains(name.toLowerCase())) {
+                headers.put(name.toLowerCase(), Collections.list(request.getHeaders(name)));
+            }
+        }
+        return HttpHeaders.of(headers, (s1, s2) -> true);
+    }
+
+    private HttpHeaders getResponseHeaders(HttpServletResponse httpServletResponse) {
+        Map<String, List<String>> headers = new LinkedHashMap<>();
+        for (String header : httpServletResponse.getHeaderNames()) {
+            Collection<String> value = httpServletResponse.getHeaders(header);
+            headers.put(header.toLowerCase(), new ArrayList<>(value));
+        }
+        headers.remove("set-cookie");
+
+        return HttpHeaders.of(headers, (s1, s2) -> true);
+    }
+
+    private String getEntityField(String entity, String field) {
+        List<String> prefixes = asList("$.", "$.xmEntity.", "$.data.");
+
+        for (String prefix : prefixes) {
+            try {
+                return JsonPath.read(entity, prefix + field);
+            } catch (Exception ex) {
+                log.info("JsonPath exception", ex);
+            }
+        }
+
+        return "";
+    }
+
+    private String getOperation(String method) {
```

Comment:
this is not the implementation we need according to reqs. It should be more flexible than it is now in TimeliNeinterceptor